airbyte-webapp/src/services/connectorBuilder/ConnectorBuilderStateService.tsx

```diff
@@ -1,71 +1,36 @@
-import React, { useContext, useEffect, useState } from "react";
+import React, { useContext, useEffect, useMemo, useState } from "react";
 
 import {
   StreamReadRequestBodyConfig,
   StreamsListReadStreamsItem,
   StreamsListRequestBodyManifest,
 } from "core/request/ConnectorBuilderClient";
+import { useAppMonitoringService } from "hooks/services/AppMonitoringService";
 
 import { useListStreams } from "./ConnectorBuilderApiService";
 
 interface Context {
   jsonManifest: StreamsListRequestBodyManifest;
   yamlIsValid: boolean;
   streams: StreamsListReadStreamsItem[];
-  selectedStream: StreamsListReadStreamsItem;
-  selectedSlice: number;
-  selectedPage: number;
+  selectedStream?: StreamsListReadStreamsItem;
   configString: string;
   configJson: StreamReadRequestBodyConfig;
   setJsonManifest: (jsonValue: StreamsListRequestBodyManifest) => void;
   setYamlIsValid: (value: boolean) => void;
   setSelectedStream: (streamName: string) => void;
-  setSelectedSlice: (sliceIndex: number) => void;
-  setSelectedPage: (pageIndex: number) => void;
   setConfigString: (configString: string) => void;
 }
 
 export const ConnectorBuilderStateContext = React.createContext<Context | null>(null);
 
-const useJsonManifest = () => {
+export const ConnectorBuilderStateProvider: React.FC<React.PropsWithChildren<unknown>> = ({ children }) => {
```

Comment:
Most of the changes here are just rearranging some code. I realized that the way I had this class set up before (with each of the various states being managed by separate hooks) had the fundamental issue that different users of this context would end up with different jsonManifest states.

Since the intention here was to have all of this context's state shared across all users of it, I moved that state code into this function directly, which seems to have fixed the problem.